angular/src/app/datacart/treetable/treetable.component.ts

```diff
@@ -323,31 +331,50 @@ export class TreetableComponent implements OnInit, AfterViewInit {
      * @param event - change event
      */
     cartChanged(which) {
-        console.log("Updating view for change in cart");
+        console.log("Updating view for change in cart contents");
+
+        // update selections due to cart additions or changes to item data
         let node: CartTreeNode = null;
         for (let item of this.dataCart.getFiles()) {
             node = this.dataTree.upsertNodeFor(item);
-            if (item.isSelected && (!this.tt || !this.tt.isSelected(node))) {
+            if (item.isSelected && (!this.tt || !this.tt.isSelected(node))) 
                 this.selectedData.push(node);
-                if (this.tt) this.tt.propagateSelectionUp(node, true);
+            if (this.tt) {
+                this.tt.propagateSelectionDown(node, item.isSelected);
+                this.tt.propagateSelectionUp(node, item.isSelected);
             }
```

Comment:
This block looks for cart items that been added or changed.  If there is a new item in the cart, `upsertNodeFor(item)` will add a (leaf) node for it in the cart.  It's section status must be matched to what the cart says:  I call `propagateSelectionDown()` to (un)select the leaf node itself and then `propagateSelectionUp()` to update its ancestors accordingly.